cardano-api/src/Cardano/Api.hs

```diff
@@ -101,21 +113,73 @@ shelleyGenSigningKey :: IO SigningKey
 shelleyGenSigningKey =
     SigningKeyShelley . Shelley.SKey <$> runSecureRandom genKeyDSIGN
 
+-- | Register a shelley staking pool.
+shelleyRegisterStakePool
+  :: ShelleyVerificationKeyHash
+  -- ^ Pool public key.
+  -> ShelleyVRFVerificationKeyHash
+  -- ^ VRF verification key hash.
+  -> ShelleyCoin
+  -- ^ Pool pledge.
+  -> ShelleyCoin
+  -- ^ Pool cost.
+  -> ShelleyStakePoolMargin
+  -- ^ Pool margin.
+  -> ShelleyRewardAccount
+  -- ^ Pool reward account.
+  -> ShelleyStakePoolOwners
+  -- ^ Pool owners.
+  -> Seq.StrictSeq Shelley.StakePoolRelay
+  -- ^ Pool relays.
+  -> Shelley.StrictMaybe Shelley.PoolMetaData
```

Comment:
We should avoid exposing these strict types and convert. Use list instead of seq, and normal maybe.